src/main/scala/com/gu/subscriptions/cas/service/SubscriptionService.scala

```diff
@@ -0,0 +1,55 @@
+package com.gu.subscriptions.cas.service
+
+import com.gu.membership.zuora.ZuoraApiConfig
+import com.gu.membership.zuora.soap.Zuora.{RatePlanCharge, Subscription, Authentication, RatePlan}
+import com.gu.membership.zuora.soap._
+import com.gu.monitoring.ZuoraMetrics
+import com.gu.subscriptions.cas.config.Configuration
+import com.gu.subscriptions.cas.model.SubscriptionExpiration
+import com.gu.subscriptions.cas.service.utils.ScheduledTask
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class SubscriptionService(zuoraClient: ZuoraClient) {
+  import spray.json._
+  import DefaultJsonProtocol._
+
+  case class SubsRequest(subscriberId: Option[String])
+  implicit val subsRequestFormat = jsonFormat1(SubsRequest)
+
+  def extractZuoraSubscriptionId(requestJson: String): Option[String] =
```

Comment:
It also extracts the value from the JSON request which is used further in the workflow